Docs Party 2024

notpeter opened the pull request:
This is a `docs_party` branch with a number of documentation improvements.

See 
- [`#docs`](https://zed.dev/channel/docs-8794/notes) (Zed Channel)
- [`#docs`](https://zed-industries.slack.com/archives/C04TX5CDT63) (Slack Channel)

A rough `To Be Documented` list can be found by searching for `TBD`:

```shell
git fetch && git checkout docs_party && git pull

# See all TBDs:
grep -R '^TBD:' docs

# See 5 random TBDs:
grep -R '^TBD:' docs | shuf -n 5
```

There are additional things that need documentation improvements in the Zed `#docs` channel.

Release Notes:

- Docs Party 2024

notpeter committed:
WIP Docs party

notpeter committed:
Docs party - Summer 2024

notpeter committed:
fix prettier.

notpeter committed:
Moar party.

notpeter committed:
More party planning committee.

myromeu commented:
@notpeter hi. Can I submit little (but so annoying) fix for outdated keybinding documention to this branch? It's not worth it to open new issue and do another PR because one word fix)

bajrangCoder committed:
Added list of Extension providing language support

bajrangCoder commented:
@\notpeter I have done some improvement in language.md , but I'm unable to push here.

Sorry for any inconveniences 

Check [this](https://github.com/bajrangCoder/zed/tree/docs_party)

notpeter committed:
More party less docs.

notpeter committed:
All party, all the time.

notpeter commented:
> @\notpeter I have done some improvement in language.md , but I'm unable to push here.

I cherry-picked your changes in d4ae81b0ca100040ecc8abfa057e535ffd1185cf. Thanks for this!

notpeter commented:
> @notpeter hi. Can I submit little (but so annoying) fix for outdated keybinding documention to this branch? It's not worth it to open new issue and do another PR because one word fix)

Commit it to a branch on a fork and I will happily cherry-pick.

notpeter committed:
Per marshall: This section can go, as these aren't finalized/released yet.

notpeter committed:
Merge branch 'main' into docs_party

notpeter committed:
Spelling fixes. Small improvements.

notpeter committed:
Linting fixes.

myromeu commented:
@notpeter completed there: https://github.com/myromeu/zed/tree/docs_party

notpeter commented:
> @notpeter completed there: https://github.com/myromeu/zed/tree/docs_party

@myromeu 
1. Can you ensure the email in your commit is associated with your GitHub account?
2. Can you sign our [Contributor Licensing Agreement](https://zed.dev/cla) (scroll down to the bottom and Sign in with Github)

notpeter committed:
Correct tasks default keybinding.

notpeter committed:
Remove .git from language extension links.

notpeter committed:
Add a bunch of Languages provided by community-supported Extensions.

notpeter committed:
Add gdscript and biome languages.

notpeter committed:
typo.

mrnugget committed:
docs: Document rust-analyzer `binary` options

Co-authored-by: Bennet <bennet@zed.dev>

notpeter committed:
Add AI Code completion docs stub.

notpeter committed:
Ai completitions edits.

notpeter committed:
Add Roc language docs.

mrnugget committed:
docs: Fix Subword motion example in docs

maxdeviant committed:
Fix references to extensions

Don't use "Zed extension", as it's redundant.

maxdeviant committed:
Add more languages to the `SUMMARY.md`

maxdeviant committed:
Refer to Prettier by its proper name

mrnugget committed:
docs: Document Ruby language servers

notpeter committed:
Add mdbook install instructions.

bennetbo committed:
start filling in ai-completions.md

mrnugget committed:
docs: Update terminal section in configuring zed

bennetbo committed:
docs: Add GoogleAI setup instructions

notpeter committed:
Link more pages into mdbook summary.md

mrnugget committed:
docs: Document theme settings

JosephTLyons committed:
Add gleam documentation

notpeter committed:
Link languages into mdbook SUMMARY.md

notpeter committed:
Drop prettier.md

JosephTLyons committed:
Include org and repo name for language tools

mikayla-maki committed:
Edit channels and collaboration, add a note about crashes on linux, fix some minor issues in getting-started and windows

co-authored-by: Jason <jason@zed.dev>

as-cii committed:
WIP: Language settings

mikayla-maki committed:
themes

co-authored-by: jason <jason@zed.dev>

maxbrunsfeld committed:
Add extensions section to the docs

Co-authored-by: Marshall <marshall@zed.dev>

JosephTLyons committed:
Update feedback-and-support.md

maxbrunsfeld committed:
Move Theme JSON structure docs under extensions section

Co-authored-by: Marshall <marshall@zed.dev>

JosephTLyons committed:
Fix command names and unify structure

maxbrunsfeld committed:
Remove stale links from development page to extensions pages

Co-authored-by: Marshall <marshall@zed.dev>

nathansobo committed:
docs: Add guide for configuring languages in Zed

Add comprehensive documentation on configuring and customizing
language support in Zed. Cover language-specific settings, file
associations, language servers, formatting, linting, syntax
highlighting, and advanced language features.

maxbrunsfeld committed:
Incorporate Tree-sitter docs into language extension docs page

Rename theme and language extension files

Co-authored-by: Marshall <marshall@zed.dev>

jvmncs committed:
refining some of the language of extensions/developing-themes.md

JosephTLyons committed:
Fix header indentation in language docs

JosephTLyons committed:
Bring python-ruff documentation up to speed

JosephTLyons committed:
Fix typos

JosephTLyons committed:
Include code fence

JosephTLyons committed:
Remove duplicate `of`s

JosephTLyons committed:
Include language annotations on more code fences

notpeter committed:
Fix gdscript.

notpeter committed:
Command palette documented elsewhere.

notpeter committed:
Add WIP fonts page.

notpeter committed:
Remove language-settings.md. Replaced by configuring-languages.md

notpeter committed:
Adding-new-languages docs redirect.

notpeter committed:
Improve docs/extensions/languages

notpeter committed:
More work on Zed fonts (unpublished).

notpeter committed:
Unpublished ai-completions.md

notpeter committed:
Improve keybinding context docs. Hide some TBDs behind comments.

notpeter committed:
More TBD comments.

osiewicz committed:
Document annotation nodes

notpeter committed:
Markdown cleanup.

notpeter committed:
More TBD fencing.

notpeter committed:
Improve repl.md, markdown.md. More TBD fences.

notpeter committed:
Improve TOML/taplo docs. More TBD fences.

notpeter committed:
Prettier and typos fixes.

notpeter committed:
Developing-extensions.md cleanup.

notpeter reviewed (commented):
I've review all 99 changed files and I think they look good on content.

Fixing links with [mdbook-linkcheck](https://crates.io/crates/mdbook-linkcheck) now.

notpeter committed:
Fix broken links.

notpeter committed:
Write-up for completions. More broken links (anchors).

notpeter committed:
Typo fix.

rgbkrk commented on docs/src/repl.md:
```diff
@@ -73,52 +76,73 @@ On MacOS, your system Python will _not_ work. Either set up [pyenv](https://gith
+- REPL (Almond) Setup Instructions https://almond.sh/docs/quick-start-install
+  - `brew install --cask temurin` (Eclipse foundation official OpenJDK binaries)
+  - `brew install coursier/formulas/coursier && cs setup`
+  - `coursier launch --use-bootstrap almond -- --install`
```
Rad that you added `coursier` and almond instructions here.
